qiskit/dagcircuit/dagcircuit.py

```diff
@@ -852,6 +852,12 @@ def _full_pred_succ_maps(self, pred_map, succ_map, input_circuit,
         return full_pred_map, full_succ_map
 
     def __eq__(self, other):
+        if (
+                self.global_phase != other.global_phase
+                or self.calibrations != other.calibrations
+        ):
+            return False
+
```

Comment:
I think we probably should remove this aspect from the backport. It needed some fixes on master and isn't strictly necessary to fix the phase tracking bugs. I think it will introduce issues for people using the 0.16.x series.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/5673/commits/c71e2b1066eaaed53554122a10a2e6f2ea55283f